### PR TITLE
use alpha channel in rgba_to_rgb

### PR DESCRIPTION
**Pull request by zzyunzhi**
### Description
Current implementation of `rgba_to_rgb` does not use alpha channel. 
The updated implementation overlays input RGBA image with a background RGB image (`(0, 0, 0)` by default). 

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>

  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?

</details>

### CODE REVIEW

**Comment by shijianjian** [comment]
Is there any references for such operation?

**Comment by zzyunzhi** [comment]
> Is there any references for such operation?

https://stackoverflow.com/questions/2049230/convert-rgba-color-to-rgb

Not sure whether this dropping the alpha channel, as originally implemented, is indeed the intended behavior and is used in other place of the codebase though. Please let me know if that is the case and I will close the PR.

File: kornia/color/rgb.py [hunk 1]
```diff
@@ -141,14 +142,15 @@ def rgba_to_rgb(image: torch.Tensor) -> torch.Tensor:
 
     # unpack channels
     r, g, b, a = torch.chunk(image, image.shape[-3], dim=-3)
+    r_bg, g_bg, b_bg = bg_rgb
 
     # compute new channels
     a_one = torch.tensor(1.0) - a
```
Comment (shijianjian) [inline kornia/color/rgb.py:148]: Will this work on GPU?
```suggestion
    a_one = torch.tensor(1.0, device=image.device, dtype=image.dtype) - a
```

**Review by shijianjian — COMMENTED** [review]
The formula seems to be correct as per the ref.

@cceyda What do you think?

File: kornia/color/rgb.py [hunk 2]
```diff
@@ -120,11 +120,12 @@ def bgr_to_rgba(image: torch.Tensor, alpha_val: Union[float, torch.Tensor]) -> t
     return rgb_to_rgba(x_rgb, alpha_val)
 
 
-def rgba_to_rgb(image: torch.Tensor) -> torch.Tensor:
+def rgba_to_rgb(image: torch.Tensor, bg_rgb: Tuple = (0, 0, 0)) -> torch.Tensor:
```
Comment (cceyda) [inline kornia/color/rgb.py:123]: `bg_rgb` reads a bit difficult. Maybe we should just name it to `background`

File: kornia/color/rgb.py (hunk 1 above)
Comment (cceyda) [reply kornia/color/rgb.py:148]: wouldn't `a_one = 1.0 - a`  work too? I thought torch smartly placed simple ops with scalars to correct device ?

**Comment by cceyda** [comment]
@shijianjian yes the formula is correct.
Adding the option to pass a background color would be pretty handy, since it is often used. 
Making me wonder if we should also add an alpha composition/blending functionality later on (maybe to kornia.enhance? or do we already have?) 

@zzyunzhi please also add tests for this~

File: kornia/color/rgb.py (hunk 2 above)
Comment (cceyda) [reply kornia/color/rgb.py:123]: Also should specify that values are expected to be in range [0,1] (or handle in code accordingly if 0,255)

File: kornia/color/rgb.py (hunk 1 above)
Comment (shijianjian) [reply kornia/color/rgb.py:148]: @zzyunzhi Can you update this line to either the code suggestion or @cceyda's? Either way  shall work, the previous line looks unsafe to me.

File: kornia/color/rgb.py (hunk 2 above)
Comment (shijianjian) [reply kornia/color/rgb.py:123]: Any naming suggestions from ```PIL``` or ```opencv```? Or naming  it as full as ```background_color```.

The whole lib shall support floating point range only for the differentiability. Probably adding a wrapper like:
https://github.com/kornia/kornia/blob/71acf455ee36f2050b7be5ea993b6db773f502eb/kornia/augmentation/utils/helpers.py#L10-L28

**Comment by shijianjian** [comment]
> @shijianjian yes the formula is correct.
> Adding the option to pass a background color would be pretty handy, since it is often used.
> Making me wonder if we should also add an alpha composition/blending functionality later on (maybe to kornia.enhance? or do we already have?)

```blend``` is more or less a PIL style function and we were mimicing the cv2 package structure. Thus, we put it as ```add_weighted``` than blending. It is already there in ```kornia.enhance```.

File: kornia/color/rgb.py (hunk 2 above)
Comment (cceyda) [reply kornia/color/rgb.py:123]: PIL or opencv doesn't directly support specifying a background color during conversion (have to do it manually with alpha blending prior or img copy/paste etc). 
[scikit-image](https://github.com/scikit-image/scikit-image/blob/2bbfce4786102a3a9e85839a52303e858346a08e/skimage/color/colorconv.py#L131-L187) does this and uses `background` name.

File: kornia/color/rgb.py [hunk 3]
```diff
@@ -141,14 +142,15 @@ def rgba_to_rgb(image: torch.Tensor) -> torch.Tensor:
...
     a_one = torch.tensor(1.0) - a
-    r_new: torch.Tensor = a_one * r + a * r
-    g_new: torch.Tensor = a_one * g + a * g
-    b_new: torch.Tensor = a_one * b + a * b
+    r_new: torch.Tensor = a_one * r_bg + a * r
+    g_new: torch.Tensor = a_one * g_bg + a * g
+    b_new: torch.Tensor = a_one * b_bg + a * b
 
-    return torch.cat([r, g, b], dim=-3)
+    return torch.cat([r_new, g_new, b_new], dim=-3)
```
Comment (cceyda) [inline kornia/color/rgb.py:153]: We should also clip values to [0,1] range with torch.clip()

File: kornia/color/rgb.py [hunk 4]
```diff
@@ -141,14 +142,15 @@ def rgba_to_rgb(image: torch.Tensor) -> torch.Tensor:
```
Comment (edgarriba) [inline kornia/color/rgb.py:142]:
```suggestion
    References:
        [1] https://en.wikipedia.org/wiki/Alpha_compositing#Alpha_blending
```
I'd add the reference

File: kornia/color/rgb.py (hunk 2 above)
Comment (edgarriba) [reply kornia/color/rgb.py:123]: `background` suits better as in `scikit-image` implementation. @shijianjian he wrapper thing should go in a separated PR

**Comment by shijianjian** [comment]
Hi @zzyunzhi, any updates on this?

**Comment by nouranali** [comment]
I'd like to work on this issue, where should I start?

**Comment by edgarriba** [comment]
@nouranali I’d take inspiration of this code and open a new PR. Make sure you read the contribution guidelines

**Comment by nouranali** [comment]
> @nouranali I’d take inspiration of this code and open a new PR. Make sure you read the contribution guidelines

Okay I will
